src/core/tests/test_documents.py

```diff
@@ -0,0 +1,20 @@
+import toga
+import toga_dummy
+from toga_dummy.utils import TestCase
+
+
+class DocumentTests(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.filename = 'path/to/document.txt'
+        self.document_type = 'path/to/document.txt'
+        self.document = toga.Document(filename=self.filename,
+                                   document_type=self.document_type,
+                                   app=toga_dummy)
+
+    def test_app(self):
+        self.assertEqual(self.filename, self.document.filename)
+
+    def test_read(self):
+        with self.assertRaises(NotImplementedError): self.document.read()
```

Comment:
No newline at EOF

Reply:
A much bigger concern than the newline is wrapping the context manager onto a single line. 

This *should* have been picked up by our CI service - we run a bunch of automated code format checks on every pull request, but for some reason, they didn't fire on this PR. Don't worry - it wasn't anything *you* did, @tiporan, but once I resolve the problem, you'll probably get a bunch of notifications about code format problems from @brutusthebee.

Reply:
Odd how I didn't notice that. Darn mobile. Good eye @freakboy3742